sklearn/linear_model/_ridge.py

```diff
@@ -1691,6 +1691,21 @@ def decision_function(self, y_predict):
 class _RidgeGCV(LinearModel):
     """Ridge regression with built-in Leave-one-out Cross-Validation.
 
+    Let's briefly describe the advantage of this formulation.
+
+    Using a naive grid-search approach with a leave-one-out cross-validation requires to
+    fit `n_samples` models to compute the prediction error for each sample. Finally, one
+    needs to repeat this process for each alpha in the grid.
```

Comment:
```suggestion
    _RidgeGCV uses a Generalized Cross-Validation for model selection. 
    
    It's an efficient approximation of leave-one-out cross-validation (LOO-CV), where instead of computing multiple models by excluding one data point at a time, it uses an algebraic shortcut to approximate the LOO-CV error, making it faster and computationally more efficient.
    
    Using a naive grid-search approach with a leave-one-out cross-validation in contrast requires to fit `n_samples` models to compute the prediction error for each sample and then to repeat this process for each alpha in the grid.
```

Would this be correct?

(Sorry for messing with the formatting.)

I have added what GCV stands for and tried to tie a bridge between the usual approach and this one, because I would have found it hard to figure out why the naive grid search is mentioned here if we hadn't talked about it before.

Can we also add, why using this algebraic trick is possible here and not anywhere else in the repo (as it seems after quick search)?

Reply:
> Can we also add, why using this algebraic trick is possible here and not anywhere else in the repo (as it seems after quick search)?

I think we can. It boils down to solving a regularized least squares problem. So it only works for problems formalized as:

![image](https://github.com/user-attachments/assets/20c0c228-bf62-4295-92df-e16322ebdbc3)

which is the minimization of the ridge problem. In practice, I think that we could use the same trick with a kernel ridge (that is not implemented in scikit-learn with the CV part).

Other linear problem cannot be formalized with the same minimization. For instance, the regularize is different in Lasso. Tree-based models are completely another family of models.